wallet/src/Pos/Wallet/Web/Methods/Restore.hs

```diff
@@ -55,47 +58,95 @@ import           UnliftIO (MonadUnliftIO)
 initialAccAddrIdxs :: Word32
 initialAccAddrIdxs = firstHardened
 
+mnemonicExists :: Text
+mnemonicExists = "Wallet with that mnemonics already exists"
+
+-- | Creates a wallet. If the wallet with the given passphrase already exists,
+-- then we return 'Left' of the wallet's ID.
 mkWallet
     :: L.MonadWalletLogic ctx m
-    => PassPhrase -> CWalletInit -> Bool -> m (EncryptedSecretKey, CId Wal)
+    => PassPhrase
+    -> CWalletInit
+    -> Bool
+    -> m (Either (CId Wal) (EncryptedSecretKey, CId Wal))
```

Comment:
I'll be tempted to wrap that `Left` side in a newtype to make the semantic clear at the type-level. Don't you think?

Reply:
hmmm, I don't know how much error-type-modification we want to do to the old wallet code. I could add a `newtype WalletAlreadyExists = WalletAlreadyExists (CId Wal)` but that'd overlap with what's in the new wallet. Or I could just use the old error type in the new code. Thoughts?